src/main/python/afp_cli/client.py

```diff
@@ -1,41 +1,50 @@
 # -*- coding: utf-8 -*-
-from __future__ import print_function, absolute_import, division
+from __future__ import absolute_import, division, print_function
 
-import requests
 import json
+
+import requests
 from requests.auth import HTTPBasicAuth
+from six import PY3, u
 
 
 class APICallError(Exception):
-    pass
+    def __str__(self, *args, **kwargs):
+        if PY3:
+            return super(APICallError, self).__str__(*args, **kwargs)
+        return self.message
+        return u(self.message).encode('unicode-escape')
 
 
 class AWSFederationClientCmd(object):
     """Class for a command line client which uses the afp api"""
 
     def __init__(self, *args, **kwargs):
-        self.username = kwargs.get('username', None)
-        self._password = kwargs.get('password', None)
+        self.username = kwargs.get('username', '')
+        self.password = kwargs.get('password', '')
+
         self.api_url = kwargs.get('api_url', None)
         self.ssl_verify = kwargs.get('ssl_verify', True)
 
     def call_api(self, url_suffix):
         """Send a request to the aws federation proxy"""
+        url_orig = '{0}{1}'.format(self.api_url, url_suffix)
+        # Workaround: request seems to mystically fail with utf8 stuff
+        url = requests.utils.requote_uri(url_orig)
```

Comment:
yeah, maybe remove the comment but I'd definitely keep the variable, so it won't have to be compiled at the parameter handover time.

actually there was more that was removed, I just run through the whole source code and removed the non relevant parts, so this is what remained.